### ISSUE TITLE
jumping to a function start should end the current function

### ISSUE THREAD

**Issue by williballenthin** [williballenthin/Lancelot#18]

**Comment by williballenthin** [comment]
prologue analysis should catch most of the targets. this needs to happen before linear/emu disassembly exploration.

thunks are an obvious and simple case. some other functions may contain more than one instruction before jumping to another function.

**Comment by williballenthin** [comment]
thunk example 5E4FCA9674F67C4D54718D4FD73C56CD @ 0x10001014

**Comment by williballenthin** [comment]
when adding a control flow edge, ensure the destination is not a function start. if so, add call graph edge instead.

when adding a function, ensure no control flow edges from "other functions" end here. if so, change them to call graph edges? how to determine if its an "other function"? traverse the whole graph? seems like that could get messy.

**Comment by williballenthin** [comment]
sound the outer function really be considered separate? maybe this is not the behavior we want.

**Comment by williballenthin** [comment]
#wontfix, this is a feature. functions are defined by their basic blocks, not start/end range. if some functions share code, thats a pretty cool compiler trick, and we shouldn't dance around it.
